File: java/unidad_01/Problema_1_07/src/Problema_1_07.java
import java.util.Scanner;
import java.util.InputMismatchException;

/*
 * El programa, dados los tres lados de un triángulo, calcula su
 * área.
 */

public class Problema_1_07 {

    public static float calcularArea(float lado01, float lado02, float lado03) {

        double s = (lado01 + lado02 + lado03) / 2;
        // Variable auxiliar

        return (float) Math.sqrt(s * (s - lado01) * (s - lado02) * (s - lado03));
        // Área de un triángulo
    }

    public static void main(String[] args) {

        float lado01, lado02, lado03;

        try (Scanner scanner = new Scanner(System.in)) {
            /*
             * Es una buena práctica cerrar los recursos adecuadamente después de usarlos
             * para liberar cualquier recurso asociado, como archivos abiertos, conexiones
             * de red o lectores de entrada.
             */

            System.out.print("1er lado: ");
            lado01 = scanner.nextFloat();

            System.out.print("2do lado: ");
            lado02 = scanner.nextFloat();

            System.out.print("3er lado: ");
            lado03 = scanner.nextFloat();

            if (lado01 <= 0 || lado02 <= 0 || lado03 <= 0) {

                System.out.println("Error: los lados deben ser mayor que cero.");
                return;
            }
        } catch (InputMismatchException e) {
            // Se produce cuando el tipo de dato ingresado por el usuario no coincide con el
            // esperado.

            System.out.println(e);
            return;
        }
        float area = calcularArea(lado01, lado02, lado03);

        System.out.printf("El área del triángulo es igual a %.4f u^2", area);
    }
}

File: python/unidad_01/problema_1_07.py
import math

'''
El programa, dados los tres lados de un triángulo, calcula su
área.
'''

def calcular_area(lado_01, lado_02, lado_03):

    s = (lado_01 + lado_02 + lado_03) / 2
    # Variable auxiliar

    return math.sqrt(s * (s - lado_01) * (s - lado_02) * (s - lado_03))
    # Área de un triángulo

def main():

    try:
        
        lado_01 = float(input("1er lado: "))

        lado_02 = float(input("2do lado: "))

        lado_03 = float(input("3er lado: "))

        if lado_01 <= 0 or lado_02 <= 0 or lado_03 <= 0:

            print("Error: los lados deben ser mayor que cero.")
            return
        
    except ValueError as e:
        # Se produce cuando una función recibe un argumento 
        # con un valor que no es válido para esa función.

        print(e)
        return
    
    area = calcular_area(lado_01, lado_02, lado_03)

    print(f"El área del triángulo es igual a {area:.4f} u^2")

if __name__ == '__main__':

    main()